pymongo/database.py

```diff
@@ -911,7 +916,34 @@ def drop_collection(
 
         if not isinstance(name, str):
             raise TypeError("name_or_collection must be an instance of str")
-
+        full_name = "%s.%s" % (self.name, name)
+        fields = None
+        if not kwargs.get("top_level"):
+            if encrypted_fields:
+                fields = encrypted_fields.get(full_name)
+            if (
+                self.client.options.auto_encryption_opts
+                and self.client.options.auto_encryption_opts._encrypted_fields_map
+            ):
+                fields = self.client.options.auto_encryption_opts._encrypted_fields_map.get(
```

Comment:
When fields exists in the `auto_encryption_opts._encrypted_fields_map` why do we still go on to run list_collections? Assuming this is supposed to work similar to schemaMap, then shouldn't we only run list_collections when `encrypted_fields` is not given and the collection doesn't exist in auto_encryption_opts._encrypted_fields_map?